packages/dataviews/src/dataviews-layouts/grid/style.scss

```diff
@@ -149,9 +154,41 @@
 .dataviews-view-grid__card:hover .dataviews-selection-checkbox,
 .dataviews-view-grid__card:focus-within .dataviews-selection-checkbox,
 .dataviews-view-grid__card.is-selected .dataviews-selection-checkbox {
+	opacity: 1;
 	top: $grid-unit-10;
 }
 
+.dataviews-view-grid__card .dataviews-view-grid__hover-actions {
+	position: absolute;
+	top: $grid-unit-05;
+	opacity: 0;
+	right: $grid-unit-05;
+	z-index: z-index(".dataviews-view-grid__card .dataviews-selection-checkbox");
+
+	.dataviews-all-actions-button {
+		background-color: $white;
+	}
+
+	@media not (prefers-reduced-motion) {
+		transition: opacity 0.1s linear;
+	}
+
+	@media (hover: none) {
+		// Show checkboxes on devices that do not support hover.
+		top: $grid-unit-05;
+	}
+}
+
+.dataviews-view-grid__card:hover .dataviews-view-grid__hover-actions,
+.dataviews-view-grid__card:focus-within .dataviews-view-grid__hover-actions {
+	opacity: 1;
+}
+
+// Include an explicit .is-hovered class to preserve hovered state when popovers are active.
+.dataviews-view-grid__card .dataviews-view-grid__hover-actions.is-hovered {
+	opacity: 1;
+}
```

Comment:
May also want to consider ensuring `opacity: 1` is used when `aria-expanded` is true on the opener button. At the moment the button will hide itself. `focus-within` doesn't work consistently because the dropdown menu content isn't rendered in the same part of the DOM.
<img width="325" height="333" alt="Screenshot 2025-08-28 at 5 37 25 pm" src="https://github.com/user-attachments/assets/5514cc1f-6730-4abf-835c-6cc3954f894f" />

I also wonder if the selection checkbox should have matching visibility logic, so all the interactive elements for an item are always visible at the same time - e.g. whenever the dropdown is open the checkbox is also visible. Though the table layout's UI doesn't work like this so maybe not. It's a very minor thing so maybe not worth sweating about!